java-agent/instrumentation/spring-webflux-5/src/main/java8/io/opentelemetry/auto/instrumentation/springwebflux/client/TracingClientResponseMono.java

```diff
@@ -28,26 +25,33 @@ public TracingClientResponseMono(
     this.exchangeFunction = exchangeFunction;
   }
 
+  private static Span getCurrentSpan() {
+    return TRACER.getCurrentSpan();
+  }
+
   @Override
   public void subscribe(final CoreSubscriber<? super ClientResponse> subscriber) {
     final Context context = subscriber.currentContext();
-    final AgentSpan parentSpan =
-        context.<AgentSpan>getOrEmpty(AgentSpan.class).orElseGet(AgentTracer::activeSpan);
+    final Span parentSpan =
+        context.<Span>getOrEmpty(Span.class).orElseGet(TracingClientResponseMono::getCurrentSpan);
```

Comment:
```suggestion
        context.<Span>getOrEmpty(Span.class).orElseGet(TRACER::getCurrentSpan);
```
then static method above is not needed